### ISSUE TITLE
Feature Request: Fuzz Counterexample Shrinking

### ISSUE THREAD

**Issue by transmissions11** [dapphub/dapptools#821]
The fuzzer seems to report the first value it finds that leads to a failing test, however this can result in some pretty confusing calldata when simpler inputs would also reproduce the failure.

Implementing a shrinking algorithm like echidna would make debugging counterexamples a lot easier

**Comment by transmissions11** [comment]
Wait doesn’t HEVM use quickcheck? iirc quickcheck has this by default

**Comment by d-xo** [comment]
hmmm yeah this might actually be easy. Any thoughts @MrChico ?

**Comment by MrChico** [comment]
Yeah, I think we have shrinking already for the normal fuzz tests, but not for the invariant tests atm. We should implement it for invariant tests. And solve the soundness issues in the replay functionality
